Analysis/Tasks/PWGCF/dptdptcorrelations.cxx

```diff
@@ -0,0 +1,407 @@
+  {
+    //    LOGF(INFO,"New collision with %d filtered tracks", ftracks.size());
+    fhCentMultB->Fill(collision.centV0M());
+    fhVertexZB->Fill(collision.posZ());
+    int acceptedevent = DPTDPT_FALSE;
+    if (IsEvtSelected(collision)) {
+      acceptedevent = DPTDPT_TRUE;
+      fhCentMultA->Fill(collision.centV0M());
+      fhVertexZA->Fill(collision.posZ());
+      //      LOGF(INFO,"New accepted collision with %d filtered tracks", ftracks.size());
+
+      for (auto& track : ftracks) {
+        /* before track selection */
+        fhPtB->Fill(track.pt());
+        fhEtaB->Fill(track.eta());
+        fhPhiB->Fill(track.phi());
+        //  no more histograms for the time being
+        //        fhEtaVsPhiB->Fill(track.phi(),track.eta());
+        //        fhPtVsEtaB->Fill(track.eta(),track.pt());
+        if (track.charge() > 0) {
+          fhPtPosB->Fill(track.pt());
+        } else {
+          fhPtNegB->Fill(track.pt());
+        }
+
+        /* track selection */
+        /* tricky because the boolean columns issue */
+        auto [asone, astwo] = AcceptTrack(track);
+        if ((asone == DPTDPT_TRUE) or (astwo == DPTDPT_TRUE)) {
+          /* the track has been accepted */
+          fhPtA->Fill(track.pt());
+          fhEtaA->Fill(track.eta());
+          fhPhiA->Fill(track.phi());
+          //  no more histograms for the time being
+          //          fhEtaVsPhiA->Fill(track.phi(),track.eta());
+          //          fhPtVsEtaA->Fill(track.eta(),track.pt());
+          if (track.charge() > 0) {
+            fhPtPosA->Fill(track.pt());
+          } else {
+            fhPtNegA->Fill(track.pt());
+          }
+        }
+        scannedtracks(asone, astwo);
+      }
+    } else {
+      for (auto& track : ftracks) {
+        scannedtracks(DPTDPT_FALSE, DPTDPT_FALSE);
+      }
+    }
+    acceptedevents(acceptedevent);
+  }
+};
+
+// Task for building <dpt,dpt> correlations
+struct DptDptCorrelationsTask {
+
+  Filter onlyacceptedevents = (aod::dptdptcorrelations::eventaccepted == DPTDPT_TRUE);
+  Filter onlyacceptedtracks = ((aod::dptdptcorrelations::trackacceptedasone == DPTDPT_TRUE) or (aod::dptdptcorrelations::trackacceptedastwo == DPTDPT_TRUE));
+
+  using FilteredTracks = soa::Filtered<soa::Join<aod::Tracks, aod::ScannedTracks>>;
+  Partition<FilteredTracks> Tracks1 = aod::dptdptcorrelations::trackacceptedasone == DPTDPT_TRUE;
+  Partition<FilteredTracks> Tracks2 = aod::dptdptcorrelations::trackacceptedastwo == DPTDPT_TRUE;
+
+  void process(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::Cents, aod::AcceptedEvents>>::iterator const& collision,
+               FilteredTracks const& tracks)
+  {
+    if (cfgProcessPairs) {
+      /* process pairs of tracks */
+      // this will be desireable
+      //      for (auto& [track1, track2] : combinations(CombinationsFullIndexPolicy(Tracks1, Tracks2))) {
```

Comment:
It doesn't work for you?

Reply:
It fails at compile time.
I'm not sure if it is because Tracks1 and Tracks2 are different partitions on the same table or because I'm using the combination policy in a wrong way.